src/common/hooks/useRestoreToken.tsx

```diff
@@ -27,6 +27,86 @@ export const getTitleEscrowCreator = async (provider: providers.Provider): Promi
   return TitleEscrowCreatorFactory.connect(creatorContractAddress, provider);
 };
 
+const validateRestoredEntities = async (
+  provider: providers.Provider | Signer,
+  address: string,
+  previousBeneficiary: string,
+  previousHolder: string,
+  transactionDetails: ContractReceipt
+): Promise<void> => {
+  const newTitleEscrow = await TitleEscrowFactory.connect(address, provider);
+  const beneficiaryDeferred = newTitleEscrow.beneficiary();
+  const holderDeferred = newTitleEscrow.holder();
+  const [beneficiary, holder] = await Promise.all([beneficiaryDeferred, holderDeferred]);
+
+  if (beneficiary !== previousBeneficiary || holder !== previousHolder)
+    throw new Error(`Token was not restored to owner and beneficiary. Tx: ${JSON.stringify(transactionDetails)}`);
+};
+
+const sendToTitleEscrow = async (
+  previousBeneficiary: string,
+  previousHolder: string,
+  provider: providers.Provider | Signer,
+  contractInstance: TradeTrustErc721,
+  tokenId: string,
+  setState: Dispatch<SetStateAction<ContractFunctionState>>
+): Promise<void> => {
+  setState("PENDING_CONFIRMATION");
```

Comment:
iff the state is set at the front, then it makes no difference that its called in the calling function before calling this?

Reply:
that way you avoid the need for `setState` and hidden coupling from the called function from the calling function

Reply:
yes, right. i have moved the PENDING_CONFIRMATION to the part before it calls either of the functions, which also removed the need for passing in setState into them.